### PR TITLE
[ITensors] Remove `NDTensors.cpu` from `dot`

### PR DESCRIPTION
**Pull request by kmp5VT**
To address the AD issue found in https://github.com/ITensor/ITensors.jl/issues/1294, remove the unnecessary call to `NDTensors.cpu`. In a previous PR we have fixed the `[]` function for order-0 tensors on GPU by using the `@allowscalar` function.

### CODE REVIEW

**Comment by mtfishman** [comment]
Thanks, could you add a simple test for running AD on `dot(...)`?

**Comment by kmp5VT** [comment]
> Thanks, could you add a simple test for running AD on `dot(...)`?

Yes, should I add the test to `ITensors` and should I loop over different devices and or element types?

**Comment by mtfishman** [comment]
Just adding a simple test to https://github.com/ITensor/ITensors.jl/blob/v0.3.52/test/ITensorChainRules/test_chainrules.jl that matches the style of the tests there is sufficient, no need to loop over element types and devices since we don't do that there anyway.

File: test/ITensorChainRules/test_chainrules.jl [hunk 1]
```diff
@@ -352,6 +352,17 @@ Random.seed!(1234)
     args = (A, B)
     test_rrule(ZygoteRuleConfig(), f, args...; rrule_f=rrule_via_ad, check_inferred=false)
   end
+
+  @testset "issue 1294" begin
```
Comment (kmp5VT) [inline test/ITensorChainRules/test_chainrules.jl:356]: @mtfishman I added this test but it fails after 3 tests in `test_rrule` with the error 
```
test_rrule: f on ITensor,ITensor: Error During Test at /Users/kpierce/.julia/packages/ChainRulesTestUtils/by7Dj/src/testers.jl:202
  Got exception outside of a @test
  Fluxes not all equal
  ```
  I am not too familiar with this `test_rrule` function is this something I should look into fixing or is there a way to modify the test to not run the one that is expected to fail for the electron case

**Comment by mtfishman** [comment]
Looks good, thanks!

**Pull request merged**